pkg/controllers/rollout/controller_test.go

```diff
@@ -198,16 +194,284 @@ func TestReconcilerHandleResourceBinding(t *testing.T) {
+			resourceOverride: &placementv1beta1.ResourceOverride{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "resource-override-2",
+					Namespace: "test-namespace",
+				},
+				Spec: placementv1beta1.ResourceOverrideSpec{
+					Placement: &placementv1beta1.PlacementRef{
+						Name:  "test-placement",
+						Scope: placementv1beta1.NamespaceScoped,
+					},
+				},
+			},
+			enqueueCRP:    true,
+			shouldEnqueue: false,
+		},
+		"test enqueue ResourceOverride with namespace scoped placement when enqueueCRP=false": {
+			resourceOverride: &placementv1beta1.ResourceOverride{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "resource-override-3",
+					Namespace: "test-namespace",
+				},
+				Spec: placementv1beta1.ResourceOverrideSpec{
+					Placement: &placementv1beta1.PlacementRef{
+						Name:  "test-placement",
+						Scope: placementv1beta1.NamespaceScoped,
+					},
+				},
+			},
+			enqueueCRP:    false,
+			shouldEnqueue: true,
+			expectedEnqueueKey: &reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: "test-namespace", Name: "test-placement"},
+			},
+		},
+		"test skip ResourceOverride with cluster scoped placement when enqueueCRP=false": {
+			resourceOverride: &placementv1beta1.ResourceOverride{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "resource-override-4",
+					Namespace: "test-namespace",
+				},
+				Spec: placementv1beta1.ResourceOverrideSpec{
+					Placement: &placementv1beta1.PlacementRef{
+						Name:  "test-placement",
+						Scope: placementv1beta1.ClusterScoped,
+					},
+				},
+			},
+			enqueueCRP:    false,
+			shouldEnqueue: false,
+		},
+		"test skip ResourceOverride without placement": {
+			resourceOverride: &placementv1beta1.ResourceOverride{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "resource-override-5",
+					Namespace: "test-namespace",
+				},
+				Spec: placementv1beta1.ResourceOverrideSpec{
+					Placement: nil,
+				},
+			},
+			enqueueCRP:    true,
+			shouldEnqueue: false,
+		},
+		"test skip a malformatted object (not ResourceOverride)": {
+			resourceOverride: &placementv1beta1.ClusterResourceSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "not-resource-override",
+				},
+			},
+			enqueueCRP:    true,
+			shouldEnqueue: false,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			queue := &controllertest.Queue{TypedInterface: workqueue.NewTypedRateLimitingQueue[reconcile.Request](workqueue.DefaultTypedItemBasedRateLimiter[reconcile.Request]())}
+			handleResourceOverride(tt.resourceOverride, queue, tt.enqueueCRP)
+			validateEnqueueBehavior(t, name, queue, tt.shouldEnqueue, tt.expectedEnqueueKey)
+		})
+	}
+}
+
+// validateEnqueueBehavior is a helper function to validate queue enqueue behavior in tests
+func validateEnqueueBehavior(t *testing.T, testName string, queue *controllertest.Queue, shouldEnqueue bool, expectedEnqueueKey *reconcile.Request) {
+	if shouldEnqueue && queue.Len() == 0 {
```

Comment:
please compare the keys directly, instead of one by one,

if you're using cmp.Equal(got, want..)

you don't need this util and the readability is better.

Reply:
Not sure I follow. Queue does not support "List()", only "Get()" by getting one resource each time.

Reply:
Use cmp.Equal instead.